src/raiden_libs/matrix.py

```diff
@@ -221,18 +220,17 @@ def follow_address_presence(self, address: Address, refresh: bool = False) -> No
     def _get_user(self, user: Union[User, str]) -> User:
         """Creates an User from an user_id, if none, or fetch a cached User """
         user_id: str = getattr(user, "user_id", user)
-        if (
-            self.broadcast_room
-            and user_id in self.broadcast_room._members  # pylint: disable=protected-access
-        ):
-            duser: User = self.broadcast_room._members[user_id]  # pylint: disable=protected-access
-
-            # if handed a User instance with displayname set, update the discovery room cache
-            if getattr(user, "displayname", None):
-                assert isinstance(user, User)
-                duser.displayname = user.displayname
-            user = duser
-        elif not isinstance(user, User):
+        for broadcast_room in self.broadcast_rooms:
+            if user_id in broadcast_room._members:  # pylint: disable=protected-access
```

Comment:
Design question. Why are you guys accessing a private attribute of the Matrix `Room` class? They got a function for that: https://github.com/matrix-org/matrix-python-sdk/blob/a469d531b6879c5687718937a0bfcb443155ce43/matrix_client/room.py#L480

Reply:
This  is actually a little quirky, as we want to do a lookup of User by user_id: get_joined_members gives us a list of Users, where _members is already a user_id: User dict. We could add a function to expose the lookup.

Reply:
Please do that.

And still as you already know accessing private attributes of a class is really bad practise and will probably come bite us in the future. On the other hand we are using a deprecated library so ... yeah .. :man_shrugging: